hudi-client/src/test/java/org/apache/hudi/table/TestCleaner.java

```diff
@@ -885,6 +888,109 @@ private void testKeepLatestCommits(boolean simulateFailureRetry, boolean enableI
         file2P0C1));
   }
 
+  @Test
+  public void testBootstrapSourceFileCleanWithKeepLatestFileVersions() throws IOException {
+    testBootstrapSourceFileClean(HoodieCleaningPolicy.KEEP_LATEST_FILE_VERSIONS);
+  }
+
+  @Test
+  public void testBootstrapSourceFileCleanWithKeepLatestCommits() throws IOException {
+    testBootstrapSourceFileClean(HoodieCleaningPolicy.KEEP_LATEST_COMMITS);
+  }
+
+  /**
+   * Test HoodieTable.clean() with Bootstrap source file clean enable.
+   */
+  @Test
+  private void testBootstrapSourceFileClean(HoodieCleaningPolicy cleaningPolicy) throws IOException {
+    HoodieWriteConfig config = HoodieWriteConfig.newBuilder().withPath(basePath).withAssumeDatePartitioning(true)
+        .withCompactionConfig(HoodieCompactionConfig.newBuilder()
+            .withCleanBootstrapSourceFileEnabled(true)
+            .withCleanerPolicy(cleaningPolicy).retainCommits(1).retainFileVersions(2).build())
+        .build();
```

Comment:
I see some private functions in this class to test cleaning based on commits and file versions, that are re-used by other tests. Is it possible to re-use some of these already created ones. May be introduce another boolean flag for bootstrap which can do additional creation of source files and later check if they are cleaned up.

Reply:
Rewrote the test part to make use of the original test code.